src/main/java/com/openlattice/shuttle/dates/JavaDateTimeHelper.java

```diff
@@ -1,146 +1,182 @@
+             }
+             return odt;
+         }
+
+        // Try parsing into OffsetDateTime with patterns
+        OffsetDateTime odt_p = parseFromPatterns(
+                date,
+                ( toParse, formatter ) -> OffsetDateTime.parse( toParse, formatter ),
+                ( local_odt, datePattern ) -> DecadeChangeHelper
+                        .fixTwoYearPatternOffsetDateTime( local_odt, datePattern ) );
+        if ( odt_p != null ) {
+            if ( shouldAddTimezone )
+                TimeZones.checkTimezonesMatch( odt_p, zoneId );
+            return odt_p;
+        }
+
+        // Try parsing into a LocalDateTime
+        LocalDateTime ldt = parseFromLocalDateTime( date );
+        if ( ldt != null )
+            return ldt.atZone( zoneId ).toOffsetDateTime();
+
+        // Try parsing into a LocalDateTime with patterns
+        LocalDateTime ldt_p = parseFromPatterns(
+                date,
+                ( toParse, formatter ) -> LocalDateTime.parse( toParse, formatter ),
+                ( local_odt, datePattern ) -> DecadeChangeHelper.fixTwoYearPatternLocalDateTime( local_odt, datePattern ) );
+        if ( ldt_p != null )
+            return ldt_p.atZone( zoneId ).toOffsetDateTime();
+
+        logger.error( "Could not parse Date Time {}", date );
+        return null;
     }
 
-    public <R> R parseWithTwoDigitYearHandling(String date, BiFunction<String, DateTimeFormatter, R> parseFunction,
+    /**
+     * Loops through datePatterns and tries to parse input
+     *
+     * @param date              - String to parse
+     * @param parseFunction     - function to use for parsing (different depending on output requirement)
+     * @param postParseFunction - function to apply after parsing, depending on successful pattern
+     */
+    public <R> R parseFromPatterns(
+            String date,
+            BiFunction<String, DateTimeFormatter, R> parseFunction,
             BiFunction<R, String, R> postParseFunction ) {
-        if (shouldIgnoreValue(date)) {
-            return null;
-        }
-        for ( String datePattern: datePatterns ){
+        for ( int i = 0; i < datePatterns.length; i++ ) {
             try {
-                DateTimeFormatter formatter = Cached.getDateFormatForString( datePattern );
+                DateTimeFormatter formatter = Cached.getDateFormatForString( datePatterns[ i ] );
                 R result = parseFunction.apply( date, formatter );
-                return postParseFunction.apply( result, datePattern );
-            } catch ( DateTimeParseException e) {
-                if ( datePattern.equals(datePatterns[datePatterns.length - 1]) ) {
-                    logger.error("Unable to parse date {}, please see debug log for additional information: {}.", date,e);
-                }
+                return postParseFunction.apply( result, datePatterns[ i ] );
+            } catch ( DateTimeParseException e ) {
+                // do nothing
```

Comment:
do we not want to log an error if `i == datePatterns.length - 1` anymore?

Reply:
Since one string can now pass multiple parsers (iso-strings + patterns), it will always throw this.